### DISCUSSION TITLE
Codable decoding of arrays-of-ints, arrays-of-strings, etc.

### DISCUSSION THREAD

**Discussion by jeffc-dev** [skiptools/skip#116] [Q&A] [answered]
Anybody know why this Codable implementation of an enum would throw an `Type mismatch: inferred type is Array<*> but Array<Int> was expected` error when trying to set itself during decode to a case with an associated value that would seem to be valid?

This works fine on iOS (`intArray` is an array of integers, as expected) but it fails with the above error on Android when setting self.

The same seems to happen with arrays of strings, doubles, and bools. The init also supports (non-array) strings, integers, etc. Those seem to work fine on both platforms.

```
public enum TestValue: Decodable {
  case numberArray([Double])
  case integerArray([Int])
  case stringArray([String])
  case boolArray([Bool])

  public init(from decoder: Decoder) throws {
    let container = try decoder.singleValueContainer()

      if let intArray = try? container.decode([Int].self) {
        self = .integerArray(intArray)
        // ...

```

**Comment by aabewhite** [comment]
Thanks for the report! We'll look into this. I believe our implementation may only support using keyed and unkeyed containers for arrays. So using an unkeyed container might be a temporary workaround if you can, but we'll try to get this corrected ASAP

(And hey, maybe do some timer work while we're at it :)

**Comment by aabewhite** [comment]
This work is complete, but it involves coordinated updates to multiple modules, so I may have to wait until I have an uninterrupted block of time tonight to push and tag it all in quick succession

**Comment by aabewhite** [comment, marked as answer]
Pushed and tagged! You should get the fix if you use File->Packages->Update to Latest in Xcode. Please let us know if you have any issues with it, and thanks again for the report

**Reply by jeffc-dev** [reply]
Thanks! I'm away from my code for a few days, but will give this a closer look soon and let you know if I run into any problems.

**Reply by jeffc-dev** [reply]
An update: I had a chance to work with this in my project and, so far, it works perfectly.

Many thanks again for the quick implementation.
